packages/wrangler/src/dev.ts

```diff
@@ -307,17 +307,18 @@ export const dev = createCommand({
 			process.exitCode = 1;
 			return;
 		}
-
+	},
+	async handler(args) {
 		if (args.remote) {
-			const isLoggedIn = await loginOrRefreshIfRequired();
+			const config = await readConfig(args, { hideWarnings: true });
+			const isLoggedIn = await loginOrRefreshIfRequired(config);
```

Comment:
In remote mode we need to connect to the API to generate the remote preview worker, which means we need to have read the config. Felt slightly awkward doing this here but couldn't think of a better way.

Reply:
Interesting... let me take a look.

Reply:
Right. I think that we can just skip this whole check. Apart from anything it would only happen once at the start, and the user could always begin in local mode and toggle to remote mid execution. So it would need to be checked there anyway.